cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -780,16 +782,14 @@ private void configProviders(KafkaConfiguration userConfig)    {
         } else {
             strimziConfigProviders = "strimzienv";
         }
-
-        if (userConfig != null
-                && !userConfig.getConfiguration().isEmpty()
-                && userConfig.getConfigOption("config.providers") != null) {
-            writer.println("# Configuration providers configured by the user and by Strimzi");
-            writer.println("config.providers=" + userConfig.getConfigOption("config.providers") + "," + strimziConfigProviders);
-            userConfig.removeConfigOption("config.providers");
-        } else {
+        
+        final String userConfigProviders = getUserConfigProviderAliases(strimziConfigProviders, userConfig);
+        if (userConfigProviders.isEmpty()) {
             writer.println("# Configuration providers configured by Strimzi");
             writer.println("config.providers=" + strimziConfigProviders);
+        } else {
+            writer.println("# Configuration providers configured by the user and by Strimzi");
+            writer.println("config.providers=" + userConfigProviders + "," + strimziConfigProviders);
```

Comment:
Doesn't this create wierd configurations such as `config.providers=,strimzienv`? Also, would it make sense to do the merging already in the separate method?